src/OpenTelemetry.Exporter.Jaeger/JaegerExporter.cs

```diff
@@ -37,36 +36,23 @@ public JaegerExporter(JaegerExporterOptions options)
             this.JaegerAgentUdpBatcher = new JaegerUdpBatcher(options);
         }
 
-        internal IJaegerUdpBatcher JaegerAgentUdpBatcher { get; }
+        internal JaegerUdpBatcher JaegerAgentUdpBatcher { get; }
 
         public override async Task<ExportResult> ExportAsync(IEnumerable<Activity> activityBatch, CancellationToken cancellationToken)
         {
-            await this.exportLock.WaitAsync().ConfigureAwait(false);
-            try
+            if (!this.libraryResourceApplied && activityBatch.Count() > 0)
             {
-                if (!this.libraryResourceApplied && activityBatch.Count() > 0)
-                {
-                    var libraryResource = activityBatch.First().GetResource();
+                var libraryResource = activityBatch.First().GetResource();
 
-                    this.ApplyLibraryResource(libraryResource ?? Resource.Empty);
+                this.ApplyLibraryResource(libraryResource ?? Resource.Empty);
 
-                    this.libraryResourceApplied = true;
-                }
+                this.libraryResourceApplied = true;
+            }
 
-                foreach (var activity in activityBatch)
-                {
-                    // avoid cancelling here: this is no return point: if we reached this point
```

Comment:
Jaeger previously ignored the cancellationToken it was passed. Spec seems to want us to have a timeout mechanism.